larray/core/axis.py

```diff
@@ -204,6 +204,10 @@ def by(self, length, step=None):
             length of groups
         step : int, optional
             step between groups. Defaults to length.
+        name_template : str, optional
```

Comment:
Honestly, I am unsure having a name_template argument is worth the maintenance cost. I think we should stop adding features nobody asked for, unless we are sure somebody will use it soon. I **know** I have been guilty of adding many such features myself but...